From 6ebfa017ebe069ddb139252e49a151ab687c4f42 Mon Sep 17 00:00:00 2001
Message-ID: <6ebfa017ebe069ddb139252e49a151ab687c4f42.1708115863.git.mjg@fedoraproject.org>
In-Reply-To: <1932a672db047da3204a445880007fcc522fa7d7.1708115863.git.mjg@fedoraproject.org>
References: <1932a672db047da3204a445880007fcc522fa7d7.1708115863.git.mjg@fedoraproject.org>
From: Julian Smith <julian.smith@artifex.com>
Date: Fri, 17 Nov 2023 17:05:06 +0000
Subject: [PATCH 07/11] scripts/wrap/cpp.py: avoid compiler error in debug
 diagnostics on Android.

In debug diagnostics, don't attempt to write a `va_list` arg to a stream.

(cherry picked from commit d888952ba9c38a0a2b0c9dec79e25c29a53e1e82)
---
 scripts/wrap/cpp.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/wrap/cpp.py b/scripts/wrap/cpp.py
index 329db25a4..0a1b16ac8 100644
--- a/scripts/wrap/cpp.py
+++ b/scripts/wrap/cpp.py
@@ -333,6 +333,8 @@ def make_fncall( tu, cursor, return_type, fncall, out, refcheck_if):
             # appears to kill std::cerr on Linux.
             out.write( f'        if ({arg.name}) std::cerr << " {arg.name}=\'" << {arg.name} << "\'";\n')
             out.write( f'        else std::cerr << " {arg.name}:null";\n')
+        elif parse.is_( arg.cursor.type, 'va_list'):
+            out.write( f'        std::cerr << " {arg.name}:va_list";\n')
         elif (0
                 or parse.is_( arg.cursor.type, 'signed char')
                 or parse.is_( arg.cursor.type, 'unsigned char')
-- 
2.44.0.rc1.222.g52f20dec8d

